src/core/Cline.ts

```diff
@@ -2897,14 +2910,6 @@ export class Cline {
 											false,
 										)
 
-										if (this.isSubTask) {
-											// tell the provider to remove the current subtask and resume the previous task in the stack (it might decide to run the command)
-											await this.providerRef
-												.deref()
-												?.finishSubTask(`new_task finished successfully! ${lastMessage?.text}`)
-											break
-										}
-
```

Comment:
Unrelated, but I noticed this while adding events - I don't think we want to pop the subtask until the attempt_completion command has finished.